### ISSUE TITLE
Front End Users/Authors are able to change plan from dashboard, without paying.

### ISSUE THREAD

**Issue by CorstoWebDesign** [aazztech/directorist#2159]
### Directorist version

Version 8.0.8

### Environment

PHP version: 8.1.30

Bricks Builder Theme

Pricing Plans Addon

### Description

When a user creates a listing in our case a free listing, this works as expected. 
The user however is able to go into their frontend dashboard and select the "Change Plan" option. They are presented with the pricing plans available and can continue. If they select a plan (say it costs 9.99) and click continue they are then redirected to the order confirmation/Checkout page. This is all correct so far but if the user decides to change their mind at this point and select the "Not Now" button and go back to the dashboard and the selected plan costing 9.99 that you haven't completed a checkout for is now applied to the listing?  

### Steps to reproduce

When a user creates a listing in our case a free listing, this works as expected. 
The user however is able to go into their frontend dashboard and select the "Change Plan" option. They are presented with the pricing plans available and can continue. If they select a plan (say it costs 9.99) and click continue they are then redirected to the order confirmation/Checkout page. This is all correct so far but if the user decides to change their mind at this point and select the "Not Now" button and go back to the dashboard and the selected plan costing 9.99 that you haven't completed a checkout for is now applied to the listing?  

### Isolating the problem

- [ ] I have reproduced this bug in localhost.
- [ ] This bug happens with a default WordPress theme active (eg. 2021, 2022 etc).

**Comment by RabbiIslamRony** [comment]
@Armanul46

**Comment by RabbiIslamRony** [comment]
No confirmed fix was found for this Pricing Plans flow. A paid plan must not be applied when the user leaves checkout through “Not Now” without completing payment. Please retest using the latest Directorist and Pricing Plans versions and provide the resulting order and listing statuses. This should remain open until the add-on flow is verified.

**Comment by RabbiIslamRony** [comment]
Verified the reported flow with Directorist 8.9.2 and Pricing Plans 4.0.1:

1. Started with an active free package.
2. Selected a paid package priced at $9.99 through Dashboard → Packages → Switch Plan.
3. Confirmed the paid package was displayed at checkout.
4. Used the checkout Cancel action to return to the dashboard without paying.

Before and after cancellation, the original free package remained active, the selected paid plan was not assigned, and both the order and payment counts remained at zero. The earlier public extension code also contains the corresponding checkout-abandonment fix in commit `10e88cb5a4becbbd71490992db4d346101d17206`.

The paid plan is now applied only after a successful checkout/package activation, so the reported unpaid plan-switch behavior is no longer reproducible.
